src/services/codefixes/addOptionalPropertyUndefined.ts

```diff
@@ -0,0 +1,137 @@
+/* @internal */
+namespace ts.codefix {
+    const addOptionalPropertyUndefined = "addOptionalPropertyUndefined";
+
+    const errorCodes = [
+        Diagnostics.Type_0_is_not_assignable_to_type_1_with_exactOptionalPropertyTypes_Colon_true_Consider_adding_undefined_to_the_type_of_the_target.code,
+        Diagnostics.Type_0_is_not_assignable_to_type_1_with_exactOptionalPropertyTypes_Colon_true_Consider_adding_undefined_to_the_types_of_the_target_s_properties.code,
+        Diagnostics.Argument_of_type_0_is_not_assignable_to_parameter_of_type_1_with_exactOptionalPropertyTypes_Colon_true_Consider_adding_undefined_to_the_types_of_the_target_s_properties.code,
+    ];
+
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const typeChecker = context.program.getTypeChecker();
+            const toAdd = getPropertiesToAdd(context.sourceFile, context.span.start, typeChecker);
+            if (!toAdd.length) {
+                return undefined;
+            }
+            const changes = textChanges.ChangeTracker.with(context, t => addUndefinedToOptionalProperty(t, toAdd));
+            return [createCodeFixAction(addOptionalPropertyUndefined, changes, Diagnostics.Add_undefined_to_optional_property_type, addOptionalPropertyUndefined, Diagnostics.Add_undefined_to_all_optional_properties)];
+        },
+        fixIds: [addOptionalPropertyUndefined],
+        getAllCodeActions: context => {
+            const { program } = context;
+            const checker = program.getTypeChecker();
+            const seen = new Map<number, true>();
+            return createCombinedCodeActions(textChanges.ChangeTracker.with(context, changes => {
+                eachDiagnostic(context, errorCodes, diag => {
+                    const toAdd = getPropertiesToAdd(diag.file, diag.start, checker);
+                    if (!toAdd.length) {
+                        return;
+                    }
+                    let untouched = true;
+                    for (const add of toAdd) {
+                        if (!addToSeen(seen, getSymbolId(add))) {
+                            untouched = false;
+                        }
+                    }
+                    if (untouched) {addUndefinedToOptionalProperty(changes, toAdd);}

```

Comment:
```suggestion
                    if (untouched) { addUndefinedToOptionalProperty(changes, toAdd); }
```

or

```suggestion
                    if (untouched) addUndefinedToOptionalProperty(changes, toAdd);
```

or

```suggestion
                    if (untouched) {
                        addUndefinedToOptionalProperty(changes, toAdd);
                    }
```

Reply:
Argh, I thought our linter had opinions on this! And would fix it for me!